src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -17,11 +17,14 @@ internal sealed class AnalyzerAnalysisContext : AnalysisContext
     {
         private readonly DiagnosticAnalyzer _analyzer;
         private readonly HostSessionStartAnalysisScope _scope;
+        
+        private readonly bool _isIOperationFeatureEnabled;
 
-        public AnalyzerAnalysisContext(DiagnosticAnalyzer analyzer, HostSessionStartAnalysisScope scope)
+        public AnalyzerAnalysisContext(DiagnosticAnalyzer analyzer, HostSessionStartAnalysisScope scope, bool isIOperationFeatureEnabled = false)
```

Comment:
Can we avoid shipping a public API with feature flag parameter? I would suggest just adding a new internal API with non optional parameter that the driver invokes.